### PR TITLE
Fix env0 configuration variable import #126

### LINKED ISSUE #126: Can't Import tf variable in scope of template
```
│ Error: could not create configurationVariable: 400 Bad Request: A property named "secret_key" already exists for organization "xxxx" on scope "BLUEPRINT"  scopeId" 6cca9d0e-656d-4b57-bd0d-3cabeb14e335
```

```
❯ tf import 'xxxx' '{  "Scope": "BLUEPRINT",  "ScopeId": "6cca9d0e-656d-4b57-bd0d-3cabeb14e335",  "name": "secret_key"}'

2021/07/17 14:43:25 [DEBUG] POST https://gitlab.com/api/v4/projects/xxxx/terraform/state/organizations_xxxx/lock
Acquiring state lock. This may take a few moments...
2021/07/17 14:43:25 [DEBUG] GET https://gitlab.com/api/v4/projects/xxxx/terraform/state/xxxx
xxxx: Importing from ID "{  \"Scope\": \"BLUEPRINT\",  \"ScopeId\": \"6cca9d0e-656d-4b57-bd0d-3cabeb14e335\",  \"name\": \"secret_key\"}"...
xxxx: Import prepared!
  Prepared env0_configuration_variable for import
xxxx: Refreshing state... [id=d84717c3-e97e-4200-94ad-a164bc2b2d61]
╷
│ Error: variable d84717c3-e97e-4200-94ad-a164bc2b2d61 not found (under this scope): <nil>
│
│
╵
```

looks legit, i think it should import it no problem, let me know if i miss anything, thanks.

--

update, ran with TF_LOG=DEBUG:

```
2021-07-26T20:22:44.175+0300 [INFO]  provider.terraform-provider-env0_v0.0.10: 2021/07/26 20:22:44 [ERROR] setting state: Invalid address to set: []string{"blueprint_id"}: timestamp=2021-07-26T20:22:44.175+0300
```

### PR DESCRIPTION
**Pull request by samuel-br**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
fixes #126 
### Solution
change all references of blueprint_id to template_id

### CODE REVIEW

**Comment by samuel-br** [comment]
the fix is not completed but the bug come from using `blueprint_id` as it is not mapping to the schema (in the schema the correspond attribute is `template_id ` ) also need to remove https://github.com/env0/terraform-provider-env0/blob/7d9a6389952589e167d8f2dc6977b8101d30a677/env0/resource_configuration_variable.go#L118
I just try to add this statement https://github.com/env0/terraform-provider-env0/blob/7d9a6389952589e167d8f2dc6977b8101d30a677/env0/resource_configuration_variable.go#L308-L310
and it solves it, we just need to decide if we patch it like that(with small sanding) or to change the schema to `blueprint_id` what is more officiant but can cause more problem to accrue

**Comment by eranelbaz** [comment]
@samuel-br i don't think that we want to change the schema as users already using this resource and the "public" name is `template` so i don't think we want to rename it to `blueprint` which is the old where we only use it in our backend

i think we want to wrap that in tests so we mitigate extra problems in the future

**Review by eranelbaz — COMMENTED** [review]
please add unit tests for that
when we fix a bug we want to add a test that makes sure we don't repeat that bug later

**Comment by samuel-br** [comment]
I didn't find a way to test import function yet, don't see example in the exist tests

**Comment by eranelbaz** [comment]
we didn't add any tests for imports
you can maybe try and use those parameters for that
```
ImportStateIdFunc, ImportState, ImportStateVerify
```

**Comment by eranelbaz** [comment]
@samuel-br 
found this example in GCP terraform provider code base
we can use the same

https://github.com/hashicorp/terraform-provider-google/blob/4ca828c2b5a8e9f1ce5c96b097aabe1d8c56873b/google/resource_assured_workloads_workload_generated_test.go#L39-L64

**Comment by samuel-br** [comment]
https://github.com/hashicorp/terraform-provider-google/wiki/Developer-Best-Practices#test-checks  
I used this article, it was very informative

**Review by eranelbaz — COMMENTED** [review]
Looking good, lets just add another use case

File: env0/resource_configuration_variable_test.go [hunk 1]
```diff
@@ -493,4 +493,60 @@ resource "%s" "test" {
 			mock.EXPECT().ConfigurationVariableDelete(configVar.Id).Times(1).Return(nil)
 		})
 	})
+
+	t.Run("import", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (samuel-br) [reply env0/resource_configuration_variable_test.go:497]: There is just import by id for this resource

[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/resource_configuration_variable_test.go:497]: we call `getConfigurationVariable` if we have `id` we use that else we query all variables in that scope and search by name

the `d.Id()` in the import function means the `terraform import` value

File: env0/resource_configuration_variable_test.go [hunk 2]
```diff
@@ -493,4 +493,60 @@ resource "%s" "test" {
...
+					ResourceName:            "env0_configuration_variable.test",
+					ImportState:             true,
+					ImportStateId:           `{  "Scope": "BLUEPRINT",  "ScopeId": "id0",  "name": "name0"}`,
+					ImportStateVerify:       true,
+					ImportStateVerifyIgnore: []string{"is_required", "is_read_only"},
+				},
+			},
+		}
+
+		configurationVariableCreateParams1 := client.ConfigurationVariableCreateParams{
```
[the remark this answers is not in the document]
Comment (samuel-br) [reply env0/resource_configuration_variable_test.go]: 6f20779a9043c7086b5413fb3f031dda33d0b3b8

File: env0/resource_configuration_variable_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuel-br) [reply env0/resource_configuration_variable_test.go]: a540222204e7cd1620585f1d11b68d8c430bfe67

**Review by eranelbaz — APPROVED** [review]

**Pull request merged**